stonecrop/src/registry.ts

```diff
@@ -9,22 +9,23 @@ export default class Registry {
 	router: Router
 	registry: Record<string, Doctype>
 	schemaLoader: (doctype: string) => Schema | Promise<Schema>
+	doctypeLoader: (doctype: string) => Doctype | Promise<Doctype>
 
-	constructor(router: Router, schemaLoader: (doctype: string) => Schema | Promise<Schema> = undefined) {
+	constructor(router: Router, doctypeLoader: (doctype: string) => Doctype | Promise<Doctype> = undefined) {
 		if (Registry._root) {
 			return Registry._root
 		}
 		Registry._root = this
 		this.name = 'Registry'
 		this.router = router
 		this.registry = {}
-		this.schemaLoader = schemaLoader
+		this.doctypeLoader = doctypeLoader
 	}
 
 	loadDoctypeSchema(doctype: Doctype) {
-		if (doctype.schemaLoader) {
-			return doctype.schemaLoader()
-		}
+		// if (doctype.schemaLoader) {
+		// 	return doctype.schemaLoader()
+		// }
```

Comment:
But you'll want it here too

Reply:
So this specific function is only responsible for loading a schema, given a doctype instance. I think this utility is no longer relevant since the schema is already available in the doctype instance itself, so we should be able to just remove this function entirely. What do you think?

Reply:
That makes sense, the registry is expecting a complete doctype so schema shouldn't be loaded separately. Let's remove it.